src/pydash/utilities.py

```diff
@@ -487,23 +622,25 @@ def memoize(func, resolver=None):
     .. versionadded:: 1.0.0
     """
 
-    def memoized(*args, **kwargs):
+    def memoized(*args: P.args, **kwargs: P.kwargs):
         if resolver:
             key = resolver(*args, **kwargs)
         else:
             key = f"{args}{kwargs}"
 
-        if key not in memoized.cache:
-            memoized.cache[key] = func(*args, **kwargs)
+        if key not in t.cast(MemoizedFunc, memoized).cache:
+            t.cast(MemoizedFunc, memoized).cache[key] = func(*args, **kwargs)
```

Comment:
Here's another example of the casting which I think obscures what's happening. Would definitely prefer separation of casting from logic. And if that's not possible, then I'd prefer not casting with whatever is needed to make the type checker happy (i.e. don't want to sacrifice readability for 100% type correctness within the internals of the functions).